src/map/Map.js

```diff
@@ -1108,6 +1119,8 @@ export var Map = Evented.extend({
 			(Browser.safari ? ' leaflet-safari' : '') +
 			(this._fadeAnimated ? ' leaflet-fade-anim' : ''));
 
+		this.setRTLDirection(DomUtil.getStyle(container, 'direction') === 'rtl');

```

Comment:
@Malvoz what do you think? Should we keep the auto detection or is a option while creating the map instance better? Both is ok for me. `L.map({ direction: 'rtl' })`

Reply:
Auto detection seems correct to me, it behaves like direction inheritance in HTML.